src/optimizer/transforms/char-class-to-meta-transform.js

```diff
@@ -10,13 +10,21 @@
  * their meta symbols equivalents.
  */
 module.exports = {
+  _hasIFlag: false,
+  _hasUFlag: false,
+  shouldRun(ast) {
```

Comment:
In this case you uses it to do initialization rather than the actual check whether the plugin should run. In this case there is `init` hook for this:

```js
init() {
  this._hasIFlag = ast.flags.includes('i');
  this._hasUFlag = ast.flags.includes('u');
}
```

Reply:
You're right! I got lazy when I realized the `init` function was called without arguments. Fixed this!

Reply:
OK, perfect!